cpp/src/sampling/detail/graph_functions.hpp

```diff
@@ -104,5 +140,45 @@ sample_edges(raft::handle_t const& handle,
              size_t fanout,
              bool with_replacement);
 
+/**
+ * @brief Randomly sample edges from the adjacency list of specified vertices
+ *
+ * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param rng_state Random number generator state
+ * @param graph_view Non-owning graph object.
+ * @param active_majors Device vector containing all the vertex id that are processed by
+ * gpus in the column communicator
+ * @param fanout How many edges to sample for each vertex
+ * @param with_replacement If true sample with replacement, otherwise sample without replacement
+ * @param invalid_vertex_id Value to use for an invalid vertex
+ * @return A tuple of device vector containing the majors, minors and weights gathered locally
```

Comment:
@return A tuple of device vectors containing the majors, minors, ids and optional weights, types and labels(of majors)